compiler/rustc_hir_analysis/src/hir_ty_lowering/bounds.rs

```diff
@@ -102,8 +102,8 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
                 seen_sized_unbound = true;
                 continue;
             }
-            // There was a `?Trait` bound, but it was not `?Sized`; warn.
-            self.dcx().span_warn(
+            // There was a `?Trait` bound, but it was not `?Sized`
+            self.dcx().span_err(
                 unbound.span,
                 "relaxing a default bound only does something for `?Sized`; \
                 all other traits are not bound by default",
```

Comment:
I don't want to block this PR on surface level things like phrasing, so feel free to ignore this and I'll submit this as a follow-up. I've never liked how this specific diagnostic is formulated, it's super verbose.

Now with this being a hard error this sentence doesn't quite make sense anymore. I'd rather see a message that's straight to the point like `` `?` can only be applied to `Sized` `` (similar to `` `~const` can only be applied to `#[const_trait]` traits `` or `` `async` bound modifier only allowed on `Fn`/`FnMut`/`FnOnce` traits ``) and defer any further explanations to diagnostic note/help messages if at all (cc error codes).

If we want to feel fancy we could provide a structured suggestion for removing the entire bound.

Reply:
How about we do this as a follow-up